services/src/main/java/org/exoplatform/task/util/TaskUtil.java

```diff
@@ -1069,6 +1069,9 @@ public static boolean hasPermissionOnField(Task task, String name, String[] valu
         Status status = statusService.getStatus(statusId);
         return status.getProject().canView(identity);
       } else if ("project".equalsIgnoreCase(name)) {
+        if(!CanEditTaskProject(task,identity)){
+          return false;
+        }
```

Comment:
This block checks if the identity can view the project (project.canView(identity)), not if the user can change the project of a task. Is it the right location to check ?